### PR TITLE
[VTA] Performance optimize, remove unnecessary contigious memory use.

### PR DESCRIPTION
**Pull request by huajsj**
Issue:
Uop maintain a cache vector to copy uop data into contigious DRAM memory for
FPGA/Simulator use, but this cache vector not get clear after FPGA/Simulator
core run, in Resnet18 case, if we printf the cache size in UopQueue::ReadBarrier
function, we can saw such cache size keep increase, this would cause
no use data copy and unnecessary contigous DRAM memory malloc.

Analysis:
This issue caused by not clear cache_ vector when do
uop_queue_.Reset().

Solution:
Override BaseQueue Reset function in UopQueue and add cache_ clear
logic.

### CODE REVIEW

**Comment by huajsj** [comment]
Hi @tmoreau89,  could you help to review this patch?

Regards
Hua

File: vta/src/runtime.cc [hunk 1]
```diff
@@ -443,9 +443,18 @@ class UopQueue : public BaseQueue<VTAUop> {
       sram_begin_ = sram_end_;
     }
   }
+
```
Comment (tmoreau89) [inline vta/src/runtime.cc]: Remove spacing (too keep it consistent with rest of function definitions

File: vta/src/runtime.cc [hunk 2]
```diff
@@ -443,9 +443,18 @@ class UopQueue : public BaseQueue<VTAUop> {
...
+  /*! \brief clear cache and reset base queue buffer.*/
+  void Reset() {
+    cache_.clear();
+    cache_idx_ = 0;
+    BaseQueue<VTAUop>::Reset();
+  }
+
```
Comment (tmoreau89) [inline vta/src/runtime.cc]: Remove spacing (too keep it consistent with rest of function definitions

File: vta/src/runtime.cc [hunk 3]
```diff
@@ -443,9 +443,18 @@ class UopQueue : public BaseQueue<VTAUop> {
...
   void AutoReadBarrier() {
     ReadBarrier();
   }
+
```
Comment (tmoreau89) [inline vta/src/runtime.cc]: Remove spacing (too keep it consistent with rest of function definitions

**Comment by tmoreau89** [comment]
Thanks for the catch @huajsj; was the fix tested in sim and hardware?

File: vta/src/runtime.cc (hunk 1 above)
Comment (huajsj) [reply vta/src/runtime.cc]: fixed.

**Comment by huajsj** [comment]
> Thanks for the catch @huajsj; was the fix tested in sim and hardware?

Hi @tmoreau89 , Thanks for the follow up, yes , the fix get tested in both sim and pynq FPGA and work fine.

Regards
Hua

**Pull request merged**